### PR TITLE
implement Butler input/output API and fix unit tests

### PR DESCRIPTION
**Pull request by n8pease**

### CODE REVIEW

File: tests/outputRoot.py [hunk 1]
```diff
@@ -154,7 +154,11 @@ def testDiffInput(self):
 
     def testBackup(self):
         mapper1 = MinMapper1(outputRoot="testOutput")
-        butler1 = dafPersist.Butler(root=None, mapper=mapper1)
+        butler1 = dafPersist.Butler(outputs=dafPersist.Repository.cfg(
+            mode='w',
+            storageCfg=dafPersist.PosixStorage.cfg(root="testOutput"),
+            mapper=mapper1)
```
Comment (ktlim) [inline tests/outputRoot.py]: Does it make sense to change the mapper so that it doesn't need to have its root separately specified?  Also, some syntactic sugar to reduce boilerplate in the common case when configuring a Butler with a PosixStorage repository might be nice.

Comment (n8pease) [inline tests/outputRoot.py]: yes, we should change CameraMapper so that it does not take a root arg. We will have to do this when we want to support more than PosixStorage (this is epic DM-6225 F16 Butler Storage & Format Refactor). For the time being I did not want to allow that scope creep into this change.
There is support to auto-populate the root keyword in mapperArgs if it's needed by the mapper __init__ method. This is a special case because it's using the outputRoot arg. 
The above mentioned code is in Repository (around line 255 currently) that will automatically add 'root' to the mapperArgs if it's not already there. 

Syntactic sugar - I'll think about that, I do agree the common case of setting up posix storage is wordy in some ways. I had introduced one idea where the args in the cfg could be a sort of 'grab bag dict (all args contained in kwargs) and various cfg __init__ methods can pull needed kwargs from that dict, but I think you did not like that idea.
This morning I was thinking about specifying root as a URI that could inform the cfg of storage type implicitly. I'm still reading about URIs, I don't know if it will work or not.
I'm open to other suggestions too (?).

File: tests/outputRoot.py [hunk 2]
```diff
@@ -183,6 +194,7 @@ def testBackup(self):
         self.assertEqual(b3Check, b3)
 
 
+
```
Comment (TallJimbo) [inline tests/outputRoot.py]: Unnecessary whitespace change.
